src/vue/forms/AssetForm.vue

```diff
@@ -359,57 +369,104 @@ export default {
+        return _merge(
+          AssetUpdateRequestRecord.fromAssetRecord(
+            requests.creationRequest.asset
+          ),
+          requests.latestUpdatableRequest || Object.assign(
+            // Request ID is assigned, because approved requests are immutable.
+            requests.latestApprovedRequest,
+            { id: ASSET_CREATION_REQUEST_ID }
           )
-          const isPreissuanceDisabled =
-            assetRecord.preissuedAssetSigner === config.NULL_ASSET_SIGNER
-
-          this.form = {
-            information: {
-              name: request.assetName,
-              code: request.assetCode,
-              maxIssuanceAmount: assetRecord.maxIssuanceAmount,
-              logo: request.logo.key
-                ? new DocumentContainer(request.logo)
-                : null,
-              policies: request.policies,
-            },
-            advanced: {
-              isPreissuanceDisabled: isPreissuanceDisabled,
-              preissuedAssetSigner: isPreissuanceDisabled
-                ? ''
-                : assetRecord.preissuedAssetSigner,
-              initialPreissuedAmount: isPreissuanceDisabled
-                ? ''
-                : assetRecord.initialPreissuedAmount,
-              terms: request.terms.key
-                ? new DocumentContainer(request.terms)
-                : null,
-            },
-          }
-        } catch (e) {
-          ErrorHandler.processWithoutFeedback(e)
+        )
+      } else {
+        return requests.creationRequest
+      }
+    },
+    async getAssetRequests (assetCode) {
+      try {
+        const { data } = await Sdk.horizon.request.getAllForAssets({
+          requestor: this.account.accountId,
+          asset: assetCode,
+        })
+        const requests = data.map(request => RecordWrapper.request(request))
+
+        const creationRequest =
+          requests.find(request => request instanceof AssetCreateRequestRecord)
+        const latestUpdateRequests = this.getLatestAssetUpdateRequests(requests
+          .filter(request => request instanceof AssetUpdateRequestRecord)
+        )
+
+        return {
+          creationRequest,
+          ...latestUpdateRequests,
         }
+      } catch (e) {
+        ErrorHandler.processWithoutFeedback(e)
       }
     },
-    async fetchAssetCreationRequest (assetCode) {
-      const { data } = await Sdk.horizon.request.getAllForAssets({
-        requestor: this.account.accountId,
-      })
-      return data.find(request => {
-        return request.details.assetCreate &&
-          request.details.assetCreate.code === assetCode
+    getLatestAssetUpdateRequests (assetUpdateRequests) {
+      const latestUpdateTime = Math.max(
+        ...assetUpdateRequests.map(request => +new Date(request.updatedAt))
+      )
+      const latestApprovedRequest = assetUpdateRequests.find(request => {
+        return request.isApproved &&
+          +new Date(request.updatedAt) === latestUpdateTime
```

Comment:
please do not use Date for comparing dates. Use moment instead, since it understands much more date formats and produces less bugs

Reply:
Fixed, used `moment`